drivers/gpio/gpio_cy8c95xx.c

```diff
@@ -0,0 +1,347 @@
+		rc = i2c_reg_write_byte(drv_data->i2c_master, cfg->i2c_slave_addr,
+					CY8C95XX_REG_PORT_SELECT, i);
+		if (rc == 0) {
+			rc = i2c_reg_write_byte(drv_data->i2c_master, cfg->i2c_slave_addr,
+						CY8C95XX_REG_DIR, dir[i]);
+		}
+	}
+	rc = i2c_burst_write(drv_data->i2c_master, cfg->i2c_slave_addr,
+			     CY8C95XX_REG_OUTPUT_DATA0, data, 3);
+
+	return rc;
+}
+
+static int cy8c95xx_config(const struct device *dev,
+			   gpio_pin_t pin,
+			   gpio_flags_t flags)
+{
+	const struct cy8c95xx_config *cfg = dev->config;
+	struct cy8c95xx_drv_data *drv_data = dev->data;
+	struct cy8c95xx_pin_state *pins = &drv_data->pin_state;
+	int rc = 0;
+	uint32_t io_flags;
+
+	/* Can't do I2C bus operations from an ISR */
+	if (k_is_in_isr()) {
+		return -EWOULDBLOCK;
+	}
+
+	/* Zephyr currently defines drive strength support based on
+	 * the behavior and capabilities of the Nordic GPIO
+	 * peripheral: strength defaults to low but can be set high,
+	 * and is controlled independently for output levels.
+	 *
+	 * SX150x defaults to high strength, and does not support
+	 * different strengths for different levels.
+	 *
+	 * Until something more general is available reject any
+	 * attempt to set a non-default drive strength.
+	 */
+	if ((flags & (GPIO_DS_ALT_LOW | GPIO_DS_ALT_HIGH)) != 0) {
+		return -ENOTSUP;
+	}
+
+	/* No open-drain support */
+	if ((flags & GPIO_SINGLE_ENDED) != 0U) {
+		return -ENOTSUP;
+	}
+
+	/* Does not support pull-up/pull-down */
+	if ((flags & (GPIO_PULL_UP | GPIO_PULL_DOWN)) != 0U) {
+		return -ENOTSUP;
+	}
+
+	/* Does not support disconnected pin, and
+	 * not supporting both input/output at same time.
+	 */
+	io_flags = flags & (GPIO_INPUT | GPIO_OUTPUT);
+	if ((io_flags == GPIO_DISCONNECTED)
+	    || (io_flags == (GPIO_INPUT | GPIO_OUTPUT))) {
+		return -ENOTSUP;
+	}
+
+	k_sem_take(&drv_data->lock, K_FOREVER);
+
+	if ((flags & GPIO_OUTPUT) != 0) {
+		pins->dir &= ~BIT(pin);
+		if ((flags & GPIO_OUTPUT_INIT_LOW) != 0) {
+			pins->data &= ~BIT(pin);
+		} else if ((flags & GPIO_OUTPUT_INIT_HIGH) != 0) {
+			pins->data |= BIT(pin);
+		}
+	} else {
+		pins->dir |= BIT(pin);
+	}
+
+	LOG_DBG("CFG %u %x : DIR %04x ; DAT %04x",
+		pin, flags, pins->dir, pins->data);
+
+	rc = write_pin_state(cfg, drv_data, pins);
+
+	k_sem_give(&drv_data->lock);
+	return rc;
+}
+
+static int port_get(const struct device *dev,
+		    gpio_port_value_t *value)
+{
+	const struct cy8c95xx_config *cfg = dev->config;
+	struct cy8c95xx_drv_data *drv_data = dev->data;
+	uint8_t pin_data[3] = { 0 };
+	uint16_t pins_data = 0;
+	int rc = 0;
+
+	/* Can't do I2C bus operations from an ISR */
+	if (k_is_in_isr()) {
+		return -EWOULDBLOCK;
+	}
+
+	k_sem_take(&drv_data->lock, K_FOREVER);
+
+	rc = i2c_burst_read(drv_data->i2c_master, cfg->i2c_slave_addr,
+			    CY8C95XX_REG_INPUT_DATA0, pin_data, sizeof(pin_data));
+
+	pins_data = pin_data[0] | ((pin_data[1] << 4) & 0xF00) |
+		    ((pin_data[2] << 12) & 0xF000);
+
+	if (rc != 0) {
+		goto out;
```

Comment:
this goto and the out: label seem a bit overdone here, why not do it the same as in port_write?